tornado/platform/asyncio.py

```diff
@@ -109,7 +109,7 @@ def initialize(  # type: ignore
         # TODO(bdarnell): consider making self.asyncio_loop a weakref
         # for AsyncIOMainLoop and make _ioloop_for_asyncio a
         # WeakKeyDictionary.
-        for loop in list(IOLoop._ioloop_for_asyncio):
+        for loop in IOLoop._ioloop_for_asyncio.copy():
```

Comment:
Hmm, this doesn't seem sufficient to fix multi-threading issues here - we could have multiple threads trying to delete the same keys from the map. I guess we're saved in practice we have concurrent loop creation only at the start of the process, and closed loops only exist at the end of the process. But while we're cleaning up here we should probably use `pop()` instead of `del` here. 

(Do you know why dict.copy() is safer than list() here? I would expect that for these two built-in types that the iteration in `list()` would be done under a single GIL critical section)

Reply:
I think catching the KeyError from the `del` is better than pop - but I'm not sure

> (Do you know why dict.copy() is safer than list() here?

Nope just that tweet :p